gateway/src/apicast/policy/keycloak_role_check/keycloak_role_check.lua

```diff
@@ -114,7 +114,10 @@ local function match_client_roles(scope, jwt)
     local client = jwt.resource_access[role.client]
 
     if not client then
-      return false
+      client = jwt.resource_access[jwt.aud]
```

Comment:
I think this should be described a bit more by comments, because it is not clear if it is correct.

Is it intentional, that when configuration says `client = "foobar"`, but there are no allowed roles for that name, it is going to fall back to `jwt.aud` ? Is it desired it is going to try both?

Is there more scripting we want do do? What about using Liquid?

Reply:
OK. 

Liquid sounds good.

If a user want to use a client application as the `client`, specify the `client` like `{{jwt.aud}}` or `{{jwt.azp}}`.

In this way, the user can select the `client` more flexibly, and this policy can detect the `client` missing.

How do you think?

Reply:
If we do liquid, then it should be working in both the client and realm roles. And for their values too. 

So following options are possible:
```json
{
"realm_roles": [
 { "name": "foo", "name_type": "plain" },
 { "name": "{{ jwt.aud }}", "name_type": "liquid" }
],
"client_roles": [
 { "name": "foo", "name_type": "plain", "client": "somename", "client_type": "plain" },
 { "name": "{{ jwt.aud }}", "name_type": "liquid", "client": "{{ jwt.azp }}", "client_type": "liquid" },
]
}
```

Later we can make this generic enough to require any value from JWT by using patterns to traverse the JWT object. But we should wait with that until we have more use cases and requirements.

For example:

```json
"whitelist": [ 
  { "path": "resource_roles.{{ jwt.azp }}.roles", "value": "{{ jwt.azp }}" }  
]
```